http/client.go

```diff
@@ -88,9 +159,18 @@ func (c *client) Send(req cmds.Request) (cmds.Response, error) {
 	} else {
 		httpReq.Header.Set(contentTypeHeader, applicationOctetStream)
 	}
-	httpReq.Header.Set(uaHeader, config.ApiVersion)
+	httpReq.Header.Set(uaHeader, c.ua)
+
+	var reqCancel func()
+	if timeoutStr, ok := req.Options[cmds.TimeoutOpt]; ok {
+		timeout, err := time.ParseDuration(timeoutStr.(string))
+		if err != nil {
+			return nil, err
+		}
+		req.Context, reqCancel = context.WithTimeout(req.Context, timeout)
+	}
 
-	httpReq.Cancel = req.Context().Done()
+	httpReq.Cancel = req.Context.Done()
```

Comment:
While we're at it, I believe the correct way to do this now is to call `httpReq.WithContext(req.Context)`.